bindings/wasm/src/did/wasm_core_document.rs

```diff
@@ -0,0 +1,448 @@
+
+  /// Returns a set of all {@link CoreService} in the document.
+  #[wasm_bindgen]
+  pub fn service(&self) -> ArrayCoreService {
+    self
+      .0
+      .service()
+      .iter()
+      .cloned()
+      .map(WasmCoreService)
+      .map(JsValue::from)
+      .collect::<js_sys::Array>()
+      .unchecked_into::<ArrayCoreService>()
+  }
+
+  // TODO: add handler for setting/removing services.
+
+  /// Returns the first {@link CoreService} with an `id` property matching the provided `query`,
+  /// if present.
+  #[wasm_bindgen(js_name = resolveService)]
+  pub fn resolve_service(&self, query: &UCoreDIDUrlQuery) -> Option<WasmCoreService> {
+    let service_query: String = query.into_serde().ok()?;
+    self
+      .0
+      .resolve_service(&service_query)
+      .cloned()
+      .map(WasmCoreService::from)
+  }
+
+  // ===========================================================================
+  // Verification Methods
+  // ===========================================================================
+
+  /// Returns a list of all {@link CoreVerificationMethod} in the DID Document.
+  #[wasm_bindgen]
+  pub fn methods(&self) -> ArrayCoreVerificationMethod {
+    self
+      .0
+      .methods()
+      .cloned()
+      .map(WasmCoreVerificationMethod::from)
+      .map(JsValue::from)
+      .collect::<js_sys::Array>()
+      .unchecked_into::<ArrayCoreVerificationMethod>()
+  }
+
+  /// Adds a new `method` to the document in the given `scope`.
+  #[wasm_bindgen(js_name = insertMethod)]
+  pub fn insert_method(&mut self, method: &WasmCoreVerificationMethod, scope: &WasmMethodScope) -> Result<()> {
+    let vm = method.0.clone();
+    let vm_gen = vm.map(CoreDID::from);
+    self.0.insert_method(vm_gen, scope.0).wasm_result()?;
+    Ok(())
+  }
+
+  /// Removes all references to the specified Verification Method.
+  #[wasm_bindgen(js_name = removeMethod)]
+  pub fn remove_method(&mut self, did: &WasmCoreDIDUrl) -> Result<()> {
+    self.0.remove_method(&did.0).wasm_result()
+  }
+
+  /// Returns a copy of the first verification method with an `id` property
+  /// matching the provided `query` and the verification relationship
+  /// specified by `scope`, if present.
+  #[wasm_bindgen(js_name = resolveMethod)]
+  pub fn resolve_method(
+    &self,
+    query: &UCoreDIDUrlQuery,
+    scope: Option<RefMethodScope>,
+  ) -> Result<Option<WasmCoreVerificationMethod>> {
+    let method_query: String = query.into_serde().wasm_result()?;
+    let method_scope: Option<MethodScope> = scope.map(|js| js.into_serde().wasm_result()).transpose()?;
+
+    let method: Option<&VerificationMethod> = self.0.resolve_method(&method_query, method_scope);
+    Ok(method.cloned().map(WasmCoreVerificationMethod))
+  }
+
+  /// Attaches the relationship to the given method, if the method exists.
+  ///
+  /// Note: The method needs to be in the set of verification methods,
+  /// so it cannot be an embedded one.
+  #[allow(non_snake_case)]
```

Comment:
Is this annotation necessary?

Reply:
I think it's necessary since `didUrl` is not snake case.